src/mrpro/utils/RandomGenerator.py

```diff
@@ -859,10 +859,11 @@ def gaussian_variable_density_samples(
             raise ValueError('n_samples must be <= (high - low)')
         n_random = n_samples - len(always_sample)
         if n_random < 0:
-            raise ValueError('more always sampled points requested than total number of samples')
+            raise ValueError('more "always sampled points" requested than total given number of samples')
         elif n_random == 0:
             return torch.tensor(always_sample).sort().values.broadcast_to(*n_batch, -1)
-        pdf = torch.exp(-torch.tensor(2.0).log() * (2 * torch.arange(low, high) / fwhm) ** 2)
+        pdf = torch.exp(-torch.tensor(2.0).log() * (2 * torch.arange(low - high // 2, low + high // 2) / fwhm) ** 2)
```

Comment:
This looks wrong.

Reply:
The fix should be in the trajectory calculator which is using this function wrong?